Assignment2/BinaryTree.java

```diff
@@ -0,0 +1,98 @@
+package Assignment2;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class BinaryTree<T> {
+	Node<T> root;
+
+	public BinaryTree() {
+
+	}
+
+	public BinaryTree(Node<T> root) {
+		this.root = root;
+	}
+
+	/**
+	 * Add a new node in the tree
+	 * @param key		the key of the new node
+	 * @param parentKey	the key of the new node's parent
+	 * @param isLeft	true if the new node will be left son, false otherwise
+	 */
+	public void addNode(T key, T parentKey, boolean isLeft) {
+		/* The tree doesn't have a root => 
+		 * Initialize root with this new node */
+		if (this.root == null) {
+			addRoot(key, true);
+			return;
+		}
+
+		/* Use BFS in order to find the parent of the node 
+		 * (assumed that there are no duplicate keys in the tree) */
+		Queue<Node<T>> queue = new LinkedList<Node<T>>();
+
+		/* Add the root in the queue */
+		queue.add(root);
+
+		Node<T> currentNode;
+		while (!queue.isEmpty()) {
+			/* Extract the first node from the queue */
+			currentNode = queue.poll();
+			
+			/* The parent of the new node was found */
+			if (currentNode.getKey() == parentKey) {
+				/* Add the new node to the left */
+				if (isLeft == true) {

```

Comment:
It would be good to have a warning here that the existing child is being overwritten (and lost). Maybe a return value?